src/managers/KeybindManager.cpp

```diff
@@ -1928,6 +1930,21 @@ void CKeybindManager::focusWindow(std::string regexp) {
     g_pCompositor->warpCursorTo(PWINDOW->middle());
 }
 
+void CKeybindManager::tagWindow(std::string args) {
+    PHLWINDOW PWINDOW = nullptr;
+    CVarList  vars{args, 0, 's', true};
+
+    if (vars.size() == 1)
+        PWINDOW = g_pCompositor->m_pLastWindow.lock();
+    else if (vars.size() == 2)
+        PWINDOW = g_pCompositor->getWindowByRegex(vars[1]);
+    else
+        return;
+
+    if (PWINDOW && PWINDOW->m_tags.applyTag(vars[0]))
+        PWINDOW->updateDynamicRules();
```

Comment:
Upon `updateDynamicRules()`, which one should I call to ensure necessary render update?

Reply:
yes